dev-tools/packaging/package_test.go

```diff
@@ -939,3 +946,69 @@ func readDockerManifest(r io.Reader) (*dockerManifest, error) {
 
 	return manifests[0], nil
 }
+
+func checkSha512PackageHash(t *testing.T, packageFile string) {
+	t.Run("check hash file", func(t *testing.T) {
+		expectedHashFile := packageFile + ".sha512"
+		require.FileExists(t, expectedHashFile, "hash file for package %q should exist with name %q", packageFile, expectedHashFile)
+
+		// calculate SHA512 hash for the file
+		hashFile, err := os.Open(expectedHashFile)
+		require.NoError(t, err, "hash file should be readable")
+
+		checksumsMap := readHashFile(t, hashFile)
+
+		packageBaseName := filepath.Base(packageFile)
+		require.Containsf(t, checksumsMap, packageBaseName, "checksum file should contain an entry for %q", packageBaseName)
+
+		// compare checksum entry with actual package hash
+		checksum := calculateChecksum(t, packageFile, sha512.New())
+
+		assert.Equalf(t, checksum, checksumsMap[packageBaseName], "checksum for file %q does not match", packageFile)
+	})
+
+}
+
+func calculateChecksum(t *testing.T, file string, hasher hash.Hash) string {
+
+	input, err := os.Open(file)
+	require.NoErrorf(t, err, "error opening input file %q", file)
+
+	defer func(input *os.File) {
+		errClose := input.Close()
+		assert.NoErrorf(t, errClose, "error closing input file %q", file)
+	}(input)
+
+	_, err = io.Copy(hasher, input)
+	require.NoError(t, err, "error reading file to calculate hash")
+
+	return hex.EncodeToString(hasher.Sum(nil))
+}
+
+// readHashFile return a map of {filename, hash} reading a .sha512 file.
+// If any line has not exactly 2 tokens separated by whitespaces, it will fail the test
```

Comment:
[Typo?]
```suggestion
// If any line has not exactly 2 tokens separated by white spaces, it will fail the test.
```

Reply:
I generally use the spelling `whitespace` when referring to blank characters like in a regular expression but I don't have a strong preference besides habit :sweat_smile:

Reply:
good point, my bad. but please add the period :)